File: backend/src/main/java/com/hardware/erp/tenant/service/TenantBankAccountService.java
package com.hardware.erp.tenant.service;

import com.hardware.erp.tenant.dto.TenantBankAccountRequest;
import com.hardware.erp.tenant.dto.TenantBankAccountResponse;
import com.hardware.erp.tenant.entity.TenantBankAccountQr;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/** Backs Shop Settings' "Bank accounts" card (CR-036) - multiple receiving accounts per shop, each with its own set of uploaded QR images, selectable per invoice. */
public interface TenantBankAccountService {

    List<TenantBankAccountResponse> list();

    TenantBankAccountResponse create(TenantBankAccountRequest request);

    TenantBankAccountResponse update(Long id, TenantBankAccountRequest request);

    void delete(Long id);

    /** Full, unmasked account number - permission-gated the same as the masked list, since this is the owner's own settings. */
    String revealAccountNumber(Long id);

    TenantBankAccountResponse addQr(Long bankAccountId, String label, MultipartFile file);

    void removeQr(Long qrId);

    TenantBankAccountQr getQrImage(Long qrId);
}

File: frontend/src/modules/settings/services/tenantBankAccountService.ts
import {
  apiDelete, apiGet, apiPost, apiPut, http,
} from '@/services/apiClient';
import type { ApiResponse } from '@/shared/types/api';
import type { TenantBankAccountRequest, TenantBankAccountResponse } from '../types';

/** Backend: tenant/controller/TenantBankAccountController.java (CR-036) */
export const tenantBankAccountService = {
  list: () => apiGet<TenantBankAccountResponse[]>('/v1/settings/bank-accounts'),

  create: (body: TenantBankAccountRequest) =>
    apiPost<TenantBankAccountResponse>('/v1/settings/bank-accounts', body),

  update: (id: number, body: TenantBankAccountRequest) =>
    apiPut<TenantBankAccountResponse>(`/v1/settings/bank-accounts/${id}`, body),

  remove: (id: number) => apiDelete(`/v1/settings/bank-accounts/${id}`),

  reveal: (id: number) => apiGet<string>(`/v1/settings/bank-accounts/${id}/reveal`),

  addQr: async (bankAccountId: number, label: string, file: File): Promise<TenantBankAccountResponse> => {
    const form = new FormData();
    form.append('label', label);
    form.append('file', file);
    const { data } = await http.post<ApiResponse<TenantBankAccountResponse>>(
      `/v1/settings/bank-accounts/${bankAccountId}/qr`, form,
      { headers: { 'Content-Type': 'multipart/form-data' } },
    );
    return data.data;
  },

  removeQr: (qrId: number) => apiDelete(`/v1/settings/bank-accounts/qr/${qrId}`),

  qrImageUrl: (qrId: number) => `/v1/settings/bank-accounts/qr/${qrId}/image`,
};
